apis/v1alpha2/grpcroute_types.go

```diff
@@ -0,0 +1,552 @@
+	// the above criteria.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=8
+	// +kubebuilder:default={{method: {type: "Exact"}}}
+	Matches []GRPCRouteMatch `json:"matches,omitempty"`
+
+	// Filters define the filters that are applied to requests that match
+	// this rule.
+	//
+	// The effects of ordering of multiple behaviors are currently unspecified.
+	// This can change in the future based on feedback during the alpha stage.
+	//
+	// Conformance-levels at this level are defined based on the type of filter:
+	//
+	// - ALL core filters MUST be supported by all implementations.
+	// - Implementers are encouraged to support extended filters.
+	// - Implementation-specific custom filters have no API guarantees across
+	//   implementations.
+	//
+	// Specifying a core filter multiple times has unspecified or
+	// implementation-specific conformance.
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	Filters []GRPCRouteFilter `json:"filters,omitempty"`
+
+	// BackendRefs defines the backend(s) where matching requests should be
+	// sent.
+	//
+	// Failure behavior here depends on how many BackendRefs are specified and
+	// how many are invalid.
+	//
+	// If *all* entries in BackendRefs are invalid, and there are also no filters
+	// specified in this route rule, *all* traffic which matches this rule MUST
+	// receive an `UNAVAILABLE` status.
+	//
+	// See the GRPCBackendRef definition for the rules about what makes a single
+	// GRPCBackendRef invalid.
+	//
+	// When a GRPCBackendRef is invalid, `UNAVAILABLE` statuses MUST be returned for
+	// requests that would have otherwise been routed to an invalid backend. If
+	// multiple backends are specified, and some are invalid, the proportion of
+	// requests that would otherwise have been routed to an invalid backend
+	// MUST receive an `UNAVAILABLE` status.
+	//
+	// For example, if two backends are specified with equal weights, and one is
+	// invalid, 50 percent of traffic MUST receive an `UNAVAILABLE` status.
+	// Implementations may choose how that 50 percent is determined.
+	//
+	// Support: Core for Kubernetes Service
+	//
+	// Support: Implementation-specific for any other resource
+	//
+	// Support for weight: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=16
+	BackendRefs []GRPCBackendRef `json:"backendRefs,omitempty"`
+}
+
+// GRPCRouteMatch defines the predicate used to match requests to a given
+// action. Multiple match types are ANDed together, i.e. the match will
+// evaluate to true only if all conditions are satisfied.
+//
+// For example, the match below will match a gRPC request only if its service
+// is `foo` AND it contains the `version: v1` header:
+//
+// ```
+// matches:
+// - method:
+//    type: Exact
+//    service: "foo"
+//   headers:
+//   - name: "version"
+//     value "v1"
+// ```
+type GRPCRouteMatch struct {
+	// Path specifies a gRPC request service/method matcher. If this field is not
```

Comment:
Comment says "Path" but field is "Method"

Reply:
Good catch, fixed in #1576.